spec/lib/que/adapters/active_record_with_lock_spec.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+RSpec.describe Que::Adapters::ActiveRecordWithLock, :active_record_with_lock do
+  subject(:adapter) do
+    described_class.new(job_connection_pool: JobRecord.connection_pool,
+                        lock_connection_pool: lock_connection_pool)
+  end
+
+  around do |example|
+    Que.adapter.tap do |old_adapter|
+      # We need this to avoid errors related to prepared statements
+      if old_adapter.class != described_class
+        Que.adapter = adapter
+        example.run
+        Que.adapter = old_adapter
+      else
+        example.run
+      end
+    end
+  end
+
+  let(:lock_connection_pool) do
+    return LockDatabaseRecord.connection_pool if ENV["ADAPTER"] == "ActiveRecordWithLock"
```

Comment:
Now have separate tests for the two adapters, I think we should use shared specs to define "Que behaviour" for all adapters, so that we only run those per adapter rather than all the specs - this feels a bit clearer than controlling this through env vars.

Non-blocking, can be a clean-up later

Reply:
Since this is required only for this Adapter I think adding a filter would be better. It gets bit tricky because of the 2 database here.
Rest all the specs are run against specific adapter.

Reply:
I have updated this to exclude all the files in adapter spec directory except for the adapter spec